### PR TITLE
hw-34 added solution

### PR DESCRIPTION
**Pull request by OlesiaKapustynska**

### CODE REVIEW

**Review by deniskichan — APPROVED** [review]

File: src/main/java/mate/academy/model/CinemaHall.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
+package mate.academy.model;
+
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "cinema_hall")
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/model/CinemaHall.java:10]:
```suggestion
@Table(name = "cinema_halls")
```

File: src/main/java/mate/academy/model/MovieSession.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+import javax.persistence.FetchType;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "movie_session")
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/model/MovieSession.java:15]:
```suggestion
@Table(name = "movie_sessions")
```
you can also change `movie` to `movies` - we will fix this

File: src/main/java/mate/academy/service/impl/MovieSessionServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    }
+
+    @Override
+    public MovieSession get(Long id) {
+        return movieSessionDao.get(id).orElseThrow();
+    }
+
+    @Override
+    public List<MovieSession> findAvailableSessions(Long movieId, LocalDate date) {
+        return movieSessionDao.findAvailableSessions(movieId,date);
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/service/impl/MovieSessionServiceImpl.java:28]:
```suggestion
        return movieSessionDao.findAvailableSessions(movieId, date);
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]
great, added minor comments

File: src/main/java/mate/academy/service/impl/MovieSessionServiceImpl.java (hunk 3 above)
Comment (OlesiaSmitiukh) [reply src/main/java/mate/academy/service/impl/MovieSessionServiceImpl.java]: done ✔️

File: src/main/java/mate/academy/model/MovieSession.java (hunk 2 above)
Comment (OlesiaSmitiukh) [reply src/main/java/mate/academy/model/MovieSession.java]: edited ✔️

File: src/main/java/mate/academy/model/CinemaHall.java (hunk 1 above)
Comment (OlesiaSmitiukh) [reply src/main/java/mate/academy/model/CinemaHall.java]: edited ✔️

File: src/main/java/mate/academy/dao/impl/MovieDaoImpl.java [hunk 4]
```diff
@@ -45,6 +45,10 @@ public Movie add(Movie movie) {
 
     @Override
     public List<Movie> getAll() {
-        return null;
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            return session.createQuery("FROM Movie ", Movie.class).getResultList();
```
[the remark this answers is not in the document]
Comment (OlesiaSmitiukh) [reply src/main/java/mate/academy/dao/impl/MovieDaoImpl.java]: done ✔️

File: src/main/java/mate/academy/Main.java [hunk 5]
```diff
@@ -1,16 +1,41 @@
...
+        CinemaHall firstHall = new CinemaHall(80, "The best cinema hall");
+        cinemaHallService.add(firstHall);
+        System.out.println(cinemaHallService.get(firstHall.getId()));
+        cinemaHallService.getAll().forEach(System.out::println);
+
+        MovieSession firstSession = new MovieSession(fastAndFurious, firstHall,
+                LocalDateTime.now());
+        movieSessionService.add(firstSession);
+        System.out.println(movieSessionService.get(firstSession.getId()));
+        movieSessionService.findAvailableSessions(1L, LocalDate.now());
```
[the remark this answers is not in the document]
Comment (OlesiaSmitiukh) [reply src/main/java/mate/academy/Main.java]: done ✔️

**Review by KseniiaMakarova — APPROVED** [review]

**Comment by OlesiaKapustynska** [comment]
I'll create a new PR

**Pull request closed without merging**